main.go

```diff
@@ -296,6 +296,7 @@ var head = []string{
 	"# encoding:",              // Ruby encoding
 	"# frozen_string_literal:", // Ruby interpreter instruction
 	"<?php",                    // PHP opening tag
+	"# syntax=",                // Dockerfile directive https://docs.docker.com/engine/reference/builder/#parser-directives
```

Comment:
Perhaps use `# syntax` (without `=`) to allow for spaces around the `=`
Also may want to add `# escape`, which is also a directive.

Reply:
> 
> 
> Perhaps use `# syntax` (without `=`) to allow for spaces around the `=`
> Also may want to add `# escape`, which is also a directive.

:+1:

> 🤔 wondering what it does if a Dockerfile has _both_ (`syntax` _and_ `escape`), although `escape` is mostly used on Windows, so not yet using BuildKit (and thus no `syntax`)

Not sure actually good point.